using-directus-as-a-baby-health-tracker/index.md

```diff
@@ -0,0 +1,362 @@
+            "type": "info",
+            "severity": "mid",
+            "priority": "P3",
+            "alias": "Low Battery",
+            "message": "The battery is low. Right now it has "+data.$trigger.payload.battery_level+"%",
+            "description": "The battery is low. Right now it has "+data.$trigger.payload.battery_level+"%"
+        }
+    }
+    // ... 
+	return rtn;
+}
+```
+
+
+The next Condition Operation within the Flow checks the value of `create_alert` and either stops here if `false` or passes the data to the final Create operation if set to `true`. This can be done easily by the following condition rule:
+
+```json
+{
+    "$last": {
+        "create_alert": {
+            "_eq": true
+        }
+    }
+}
+```
+You can build in several other logic like checking if there is an open alert already and stuff like this. To keep it simple we're skipping this at the moment. (And trust OpsGenie to not raise two alerts for the same incoming data)
+
+Within the create operation, the following payload is used to create the Alert item:
+```json
+{
+    "category": "{{exec_alert_logic.alert_data.category}}",
+    "type": "{{exec_alert_logic.alert_data.type}}",
+    "severity": "{{exec_alert_logic.alert_data.severity}}",
+    "priority": "{{exec_alert_logic.alert_data.priority}}",
+    "alias": "{{exec_alert_logic.alert_data.alias}}",
+    "message": "{{exec_alert_logic.alert_data.message}}",
+    "description": "{{exec_alert_logic.alert_data.description}}"
+}
+```
+#### Escalate the alert data to OpsGenie    
+Once the alert is created within Directus, of course, it has to be created within OpsGenie as well. For this – again a new Flow is needed. This time we have to use the Timeout operation as OpsGenie queues all incoming requests and does not provide the result immediately.
+
+!(Pasted image 20240304175218.png)
+Once triggered, we read out all the data first and pass the needed data to the OpsGenie API with a user key and the JSON data from the alert collection. As the API does not return the actual result but only an ID of the incoming task, we now wait 2 seconds before requesting the current status of our queued task with the 2nd web request. The request URL gets the returning ID from the 1st web request: 
+!(Pasted image 20240304175856.png)
+
+This request will return the current status and therefore all the data (including an ID) of this alert. With the update operation, we saved the alert ID to be able to refer to the OpsGenie alert if needed.
+In the moment an actual alert is created within OpsGenie, any connected system is yelling for attention, just as configured in the tool.
+If Directus tries to create a new alert with the same data (in case the battery or oxygen level is still on the same level) no new alert is created by the design of OpsGenie. But it's hard to miss any alert as it sends SMS, critical push notifications or even calling a phone number if needed and no reaction is noticed. 
```

Comment:
```suggestion
#### Escalating to OpsGenie    

Once the alert is created within Directus, of course, it has to be created within OpsGenie as well. For this – again a new Flow is needed. This time we have to use the Timeout operation as OpsGenie queues all incoming requests and does not provide the result immediately.

!(Pasted image 20240304175218.png)

Once triggered, we read out all the data first and pass the needed data to the OpsGenie API with a user key and the JSON data from the alert collection. As the API does not return the actual result but only an ID of the incoming task, we now wait 2 seconds before requesting the current status of our queued task with the 2nd web request. The request URL gets the returning ID from the 1st web request: 

!(Pasted image 20240304175856.png)

This request will return the current status and therefore all the data (including an ID) of this alert. With the update operation, we saved the alert ID to be able to refer to the OpsGenie alert if needed.

In the moment an actual alert is created within OpsGenie, any connected system is yelling for attention, just as configured in the tool.

If Directus tries to create a new alert with the same data (in case the battery or oxygen level is still on the same level) no new alert is created by the design of OpsGenie. But it's hard to miss any alert as it sends SMS, critical push notifications, or even calling a phone number if needed and no reaction is noticed. 
```